js/apps/admin-ui/src/user/CreateUser.tsx

```diff
@@ -66,15 +94,15 @@ export default function CreateUser() {
       />
       <PageSection variant="light" className="pf-u-p-0">
         <UserProfileProvider>
-          <FormProvider {...userForm}>
-            <PageSection variant="light">
-              <UserForm
-                realm={realmRepresentation}
-                onGroupsUpdate={setAddedGroups}
-                save={save}
-              />
-            </PageSection>
-          </FormProvider>
+          <PageSection variant="light">
+            <UserForm
+              form={form}
```

Comment:
Why did you pass in the form instead?

Reply:
Type safety, this way we can be sure that the form we are dealing with is in fact provided and of the correct type.